### ISSUE TITLE
Network error when enabling PROJ_NETWORK

### ISSUE THREAD

**Issue by ktheijs** [OSGeo/gdal#6639]
## Expected behavior and actual behavior.

I've enabled PROJ_NETWORK but it throws an exception with the message 'network error'.
I have tried by doing `Osr.SetPROJEnableNetwork(true);`

and also by altering the proj7/proj.ini file by uncommenting the 'network = on' line. It looks like this:

```
[general]
; Lines starting by ; are commented lines.
;

; Network capabilities disabled by default.
; Can be overridden with the PROJ_NETWORK=ON environment variable.
network = on

; Can be overridden with the PROJ_NETWORK_ENDPOINT environment variable.
cdn_endpoint = https://cdn.proj.org

cache_enabled = on

cache_size_MB = 300

cache_ttl_sec = 86400

; Transverse Mercator (and UTM)  default algorithm: auto, evenden_snyder or poder_engsager
; * evenden_snyder is the fastest, but less accurate far from central meridian
; * poder_engsager is slower, but more accurate far from central meridian
; * default will auto-select between the two above depending on the coordinate
;   to transform and will use evenden_snyder if the error in doing so is below
;   0.1 mm (for an ellipsoid of the size of Earth)
tmerc_default_algo = poder_engsager
```

I don't get any other message.

## Operating system

Windows 11

## GDAL version and provenance

GDAL 3.5.3 from https://www.gisinternals.com/query.html?content=filelist&file=release-1930-x64-gdal-3-5-3-mapserver-8-0-0.zip
I'm using C#.

**Comment by rouault** [comment]
Please provide more details on how to replicate the error, in particular which transformation you are attempting to.
I would also suggest that you try first with the cs2cs or cct PROJ command line utilities.

**Comment by ktheijs** [comment]
Any transformation that needs a file from https://cdn.proj.org/ fails.

I've tried the following with cs2cs
`echo 45 2 0 | cs2cs "WGS 84 + EGM96 height" "WGS 84 / UTM zone 31N" `

The result is the following when I turn network off in the proj.ini file:
`421184.70	4983436.77 0.00`

The result it this when I turn network on in the proj.ini file:
`*	* inf`

I've also tried running projsync to see if it was possible to make a connection.
`projsync --endpoint https://cdn.proj.org/ --all`

But the result is this
```
Downloading from https://cdn.proj.org into C:\Users\KoenTheijs\AppData\Local/proj
Cannot download files.geojson
```

I found this article online that might be the reason: https://github.com/OSGeo/PROJ/issues/2285
Do you know if libcurl is compiled with ssl support in the version provided here: https://www.gisinternals.com/query.html?content=filelist&file=release-1930-x64-gdal-3-5-3-mapserver-8-0-0.zip ?

**Comment by rouault** [comment]
> Do you know if libcurl is compiled with ssl support in the version provided here

no, I don't know, but this seems to be a defect of that build. You may file your issue at https://github.com/gisinternals/buildsystem/

**Comment by ktheijs** [comment]
I have made some progress. The build doesn't seem to be the problem.

If I set the environment variable "CURL_CA_BUNDLE" to the "curl-ca-bundle.crt" file and run cs2cs.exe, everything works fine. The files are fetched , added to the local cache and the coordinates are transformed correctly.

However, when I try to use it in GDAL I still get the same network error issue. I've tried to set the environment variable.
`Environment.SetEnvironmentVariable("CURL_CA_BUNDLE", GdalPathHelper.GdalCurlCertificateFileFullName);`

And I've tried to set the gdal config option 
`Gdal.SetConfigOption("CURL_CA_BUNDLE", GdalPathHelper.GdalCurlCertificateFileFullName);`

Both approaches don't seem to do anything. Is there another configuration I should set?

Here is a code snippet of what I try to do. Note that this code should be fine as it works fine if I have the grid files stored locally.

```
string epsg4326 = @"GEOGCRS[""WGS 84"",DATUM[""World Geodetic System 1984"",ELLIPSOID[""WGS 84"",6378137,298.257223563,LENGTHUNIT[""metre"",1]]],PRIMEM[""Greenwich"",0,ANGLEUNIT[""degree"",0.0174532925199433]],CS[ellipsoidal,2],AXIS[""geodetic latitude (Lat)"",north,ORDER[1],ANGLEUNIT[""degree"",0.0174532925199433]],AXIS[""geodetic longitude (Lon)"",east,ORDER[2],ANGLEUNIT[""degree"",0.0174532925199433]],USAGE[SCOPE[""unknown""],AREA[""World""],BBOX[-90,-180,90,180]],ID[""EPSG"",4326]]";
SpatialReference epsg4326Reference = new SpatialReference(epsg4326);
epsg4326Reference.SetAxisMappingStrategy(AxisMappingStrategy.OAMS_TRADITIONAL_GIS_ORDER);
epsg4326Reference.PromoteTo3D($"epsg4326 (Promoted to 3D)");

SpatialReference epsg4326_5773Reference = new SpatialReference(epsg4326);
epsg4326_5773Reference.SetAxisMappingStrategy(AxisMappingStrategy.OAMS_TRADITIONAL_GIS_ORDER);
epsg4326_5773Reference.SetVertCS("EGM96 height", "EGM96 geoid", 2005);

double x = 5.73340279;
double y = 50.6086142;
double z = 299.16;
CoordinateTransformation transform = new CoordinateTransformation(epsg4326Reference, epsg4326_5773Reference);
double[] transformedCoordinates = new double[3];
transform.TransformPoint(transformedCoordinates, x, y, z);
```

**Comment by rouault** [comment]
you can set the ``ca_bundle_path`` key in proj.ini : cf https://proj.org/resource_files.html#proj-ini

**Comment by ktheijs** [comment]
That also doesn't seem to work. I assume because the GDAL build on gistinternals uses PROJ 7.2. This key seems to be added in PROJ 9.0 according to the link you've sent me.

Do you know another way to configure this?

**Comment by rouault** [comment]
> Do you know another way to configure this?

define the environment variable *before* you start C# as Environment.SetEnvironmentVariable() or  Gdal.SetConfigOption() have no effects on PROJ when it calls getenv() to read CURL_CA_BUNDLE.

But asking in https://github.com/gisinternals/buildsystem/ to update PROJ of gistinternals to PROJ 9.1 is certainly a more future proof solution

**Comment by ktheijs** [comment]
Thank you for the input. It works if I set the environment variable before starting the app.

Just as a small follow up question, are there plans to update the PROJ libraries to version 9.0 any time soon?
